### DISCUSSION TITLE
Configuring different s3_resources using different credentials

### DISCUSSION THREAD

**Discussion by szalai1** [dagster-io/dagster#3096] [Q&A] [answered]
Hi,

I need access to another company's s3 bucket.
 I have their credentials to access the bucket, is there any way to specify an `s3_resource` using different credentials or I have to use boto3 directly?

**Comment by alangenfeld** [comment, marked as answer]
The S3 resource is just a small amount of code around instantiating `boto3` https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-aws/dagster_aws/s3/utils.py#L32-L51

At this time, we just defer to `boto3` to auth via env vars or files and do not have any way to pass in creds. You could send a PR to add that capability, likely by adding an optional config section https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-aws/dagster_aws/s3/resources.py#L7-L31 or write your own resource def.

**Reply by szalai1** [reply]
Thanks a lot, I've been planning to contribute something for a long time, this can be a good first step.
